libs/git_wrapper/OGit_wrapper.ml

```diff
@@ -0,0 +1,159 @@
+(* Austin Theriault
+ *
+ * Copyright (C) Semgrep, Inc.
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public License
+ * version 2.1 as published by the Free Software Foundation, with the
+ * special exception on linking described in file LICENSE.
+ *
+ * This library is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the file
+ * LICENSE for more details.
+ *)
+
+(* Commentary *)
+(* This is like git_wrapper but using pure ocaml https://github.com/mirage/ocaml-git. *)
+(* That way we don't have to shell out *)
+
+(*****************************************************************************)
+(* Prelude *)
+(*****************************************************************************)
+
+module Store = Git_unix.Store
+module Hash = Store.Hash
+module Value = Git.Value.Make (Hash)
+module Commit = Git.Commit.Make (Hash)
+module Tree = Git.Tree.Make (Hash)
+module Blob = Git.Blob.Make (Hash)
+module User = Git.User
+
+let tags = Logs_.create_tags [ __MODULE__ ]
+
+(*****************************************************************************)
+(* Types *)
+(*****************************************************************************)
+type object_table = (Value.hash, Value.t) Hashtbl.t
```

Comment:
re exporting types here, but maybe we just reexport the modules?

Reply:
I think you should re-export everything users need. imo if we have a wrapper users shouldn't be able to use the library directly.